paddle/fluid/operators/controlflow/conditional_block_op.cc

```diff
@@ -65,6 +69,12 @@ class ConditionalBlockOp : public ConditionalOp {
       scopes->resize(1);
       scopes->front() = &scope.NewScope();
       auto &cur_scope = *scopes->front();
+#ifdef PADDLE_WITH_MKLDNN
+      // (jczaja) Executor on being destroyed clears oneDNN cache and
+      // reset registered model data layout. This is unwanted for nested
+      // Executors (executors declared inside control ops)
+      platform::DontClearMKLDNNCache(dev_place);
```

Comment:
Can this function use `enable_if_CPU` to avoid many #ifdef PADDLE_WITH_MKLDNN, Cause Baidu is suggesting to not use #ifdef PADLLE_WITH_MKLDNN recently. Maybe this can be considered in later refactoring.